sdk/eventhubs/azure-messaging-eventhubs/src/main/java/com/azure/messaging/eventhubs/EventHubProducerAsyncClient.java

```diff
@@ -248,6 +263,71 @@ public String getEventHubName() {
                 }));
     }
 
+    /**
+     * Creates an {@link ObjectBatch} that can fit as many serialized objects as events as the transport allows.
+     * @param objectType type of object in the batch
+     * @param <T> object type
+     *
+     * @return A new {@link ObjectBatch} that can fit as many serialized objects as events as the transport allows.
+     */
+    public <T> Mono<ObjectBatch<T>> createBatch(Class<T> objectType) {
+        return createBatch(objectType, DEFAULT_BATCH_OPTIONS);
+    }
+
+    /**
+     * Creates an {@link ObjectBatch} configured with the options specified.
+     *
+     * @param objectType type of object in the batch
+     * @param <T> object type
+     * @param options A set of options used to configure the {@link ObjectBatch}.
+     * @return A new {@link ObjectBatch} that can fit as many events as the transport allows.
+     * @throws NullPointerException if {@code options} is null.
+     */
+    public <T> Mono<ObjectBatch<T>> createBatch(Class<T> objectType, CreateBatchOptions options) {
+        if (objectType == null) {
+            return monoError(logger, new IllegalArgumentException("'objectType' cannot be null."));
+        }
+        if (serializer == null) {
+            return monoError(logger,
+                new NullPointerException("No serializer set for performing object serialization for ObjectBatch."));
+        }
+        if (options == null) {
+            return monoError(logger, new NullPointerException("'options' cannot be null."));
+        }
+
+        Mono<ObjectBatch<T>> optionsError = validateBatchOptions(options);
+        if (optionsError != null) {
+            return optionsError;
+        }
+
+        final String partitionKey = options.getPartitionKey();
+        final String partitionId = options.getPartitionId();
+        final int batchMaxSize = options.getMaximumSizeInBytes();
+
+        return getSendLink(partitionId)
+            .flatMap(link -> link.getLinkSize()
+                .flatMap(size -> {
```

Comment:
Should this be in a single chain instead of nesting the flatMap operators?

Reply:
Yeah I grabbed this one from existing batch creation code, looks like link ref is used for some other args.